pkg/sbom/io/encode.go

```diff
@@ -283,6 +288,14 @@ func (e *Encoder) resultComponent(root *core.Component, r types.Result, osFound
 		component.Type = core.TypeApplication
 	}
 
+	// try to look for BOM-ref for this component
+	for _, c := range e.components {
+		if c.Name == component.Name && c.Type == component.Type {
```

Comment:
IIRC we correctly store BomRefs for the `Root` component and packages.
Do we need this check for all types? maybe we can do this check only for `Application` and `OperatingSystem` types?

Reply:
done